optional-options-docs.md

```diff
@@ -0,0 +1,72 @@
+# Tricks and traps when using options with optional values
+
+There potential challenges using options with optional values. They seem quite attractive and the README used to use them more than options with require values but in practice, they are a bit tricky and aren't a free choice.
+
+## Parsing ambiguity
+
+There is parsing ambiguity when using option as boolean flag and also having it accept operands and subcommands.
+
+```
+program.command('example')
+      .option("-o, --option [optionalValue]")
+      .command('brew')
+      .action(() => {
+        console.log("example brew");
+      })
+
+program.parse(process.argv);
+
+if (program.option) console.log(`option: ${program.option}`);
+```
+
+```
+$ example -o
+option: true
+$ example -o thisValueIsPassedToOption
+option: thisValueIsPassedToOption
+$ example -o brew
+option: brew
+$ example -o nextArg
+option: nextArg
+```
+
+For example, the user may intend `brew` to be passed as a subcommand. Instead, it will be read as the passed in value for `-o`. Likewise, the user may intend `nextArg` to be passed as an argument but it too, will be read as the passed in value for `-o`
+
+### Possible workarounds
+
+To reduce such ambiguity, you can do the following:
+
+1. always use `--` before operands
+2. add your options after operands
```

Comment:
The posix convention is that options always come before operands. The GNU utility convention allows options to come before or after the operands. Commander follows the GNU convention and allows options before or after the operands. So by putting the options last, the option values do not get confused with the operands.

So instead of writing for a `--sugar` option taking an option count:

    brew --sugar -- tea

you can write without needing the dash-dash:

    brew tea --sugar